src/modules/questionnaires/questionnaire_service.ts

```diff
@@ -35,6 +38,11 @@ export class QuestionnaireService {
 
   public findById = async (id: string) => {
     const item: any = await this.questionnaireRepository.findById(id)
+    if (!item)
+      throw new HttpError(
+        httpStatus.NOT_FOUND,
+        lang.__('error.exists', { entity: 'questionnaire', id })
+      )
```

Comment:
kang @ayocodingit apakah kasus seperti ini selalu dikasih kondisional untuk handle http error not found? apakah memungkinkan untuk semua entity ketika not found diarahkan ke exception tanpa harus menulis kode kondisional ini ?

Reply:
iya betul kang, semuanya akan masuk dalam exception, menulis kode throw nya kan emng harus di tulis kang agar masuk dalam exception

Reply:
sama halnya dengan laravel menggunakan firstOrFail(), cuman secara manualnya gini kang @muhamadabduh

Reply:
nah berarti kode throw httpError itu ditulisnya apakah di setiap query atau tidak? 
 sy cek find by id yang di village handler ini ga pakai pengecekan kondisi not found [village_handler.ts](https://github.com/jabardigitalservice/dedi-core-service/blob/main/src/modules/villages/village_handler.ts#L34)

Reply:
> nah berarti kode throw httpError itu ditulisnya apakah di setiap query atau tidak? sy cek find by id yang di village handler ini ga pakai pengecekan kondisi not found [village_handler.ts](https://github.com/jabardigitalservice/dedi-core-service/blob/main/src/modules/villages/village_handler.ts#L34)

iya betul kang, balik lagi pada layer service sendiri digunakan untuk bussines logic, jadi pengecekan jika null maka di throw 404, nah kenapa di handler nya akan langsung pada catch next(error), nah next(error) ini akan langsung menuju exception kang,

Reply:
> > nah berarti kode throw httpError itu ditulisnya apakah di setiap query atau tidak? sy cek find by id yang di village handler ini ga pakai pengecekan kondisi not found [village_handler.ts](https://github.com/jabardigitalservice/dedi-core-service/blob/main/src/modules/villages/village_handler.ts#L34)
> 
> iya betul kang, balik lagi pada layer service sendiri digunakan untuk bussines logic, untuk layer repository hanya menangani query saja, jadi pengecekan jika null maka di throw 404, nah kenapa di handler nya akan langsung pada catch next(error), nah next(error) ini akan langsung menuju exception kang,

Reply:
oh ya siap kang, cukup clear buat saya. sebetulnya struktur projectnya ini menarik sekali jadi kepo, dibagi-bagi per module dan per module nya itu banyak layernya. 
saran diadakan review terkait struktur project/arsitektur dedicore ini di weekly atau ayocode talk.

Reply:
> oh ya siap kang, cukup clear buat saya. sebetulnya struktur projectnya ini menarik sekali jadi kepo, dibagi-bagi per module dan per module nya itu banyak layernya. 
> saran diadakan review terkait struktur project/arsitektur dedicore ini di weekly atau ayocode talk 

Sebenernya sudah aku jadiin bahan ayo talk pertama, mungkin nanti akan aku coba sisipkan lagi jika ada kesempatan lainnya 😁